### PR TITLE
[serve] Fix `test_batching` on Python 3.11 and split into unit/integration

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

Test was failing on Python 3.11 due to passing coroutines to `asyncio.wait` (see https://github.com/ray-project/ray/pull/40637).

I also split out the unit & integration pieces into separate files and removed some sleeps in the unit tests.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by edoakes** [comment]
Marked as a draft because I need to merge https://github.com/ray-project/ray/pull/40637 first.

**Review by shrekris-anyscale — APPROVED** [review]
Nice work!

**Review by sihanwang41 — COMMENTED** [review]
Hi Ed, i am seeing `sleep` in the unit/test_batching.py. is it required?

**Comment by edoakes** [comment]
@sihanwang41 the tests are copy-pasted. One of the tests was quite slow so I refactored it to remove the sleeps. Let me take a look at the remaining ones.

**Comment by edoakes** [comment]
The tests are passing in `1.56s` so not going to block merging on refactors

**Review by GeneDer — APPROVED** [review]
:shipit:

File: python/ray/serve/tests/unit/test_batching.py [hunk 1]
```diff
@@ -0,0 +1,592 @@
...
+
+    for batch_size in range(1, 7):
+        tasks = [
+            get_or_create_event_loop().create_task(batching_example.handle_batch(1))
+            for _ in range(batch_size)
+        ]
+
+        done, _ = await asyncio.wait(tasks, return_when="ALL_COMPLETED")
+        assert set({task.result() for task in done}) == {batch_size}
+        time.sleep(0.05)
```
Comment (sihanwang41) [inline python/ray/serve/tests/unit/test_batching.py:44]: lol, do you have know why we need to sleep here?

**Review by sihanwang41 — APPROVED** [review]

**Comment by edoakes** [comment]
Python 3.11 tests passing now :)

**Pull request merged**